File: app/src/main/java/com/amicly/alamofoursquare/common/scheduler/SchedulerProvider.java
package com.amicly.alamofoursquare.common.scheduler;

import io.reactivex.Scheduler;

/**
 * Created by darrankelinske on 2/1/18.
 */

public interface SchedulerProvider {

    Scheduler mainThread();

    Scheduler computation();

    Scheduler io();
}

File: app/src/main/java/com/amicly/alamofoursquare/common/scheduler/SchedulerProvider.kt
package com.amicly.alamofoursquare.common.scheduler

import io.reactivex.Scheduler

/**
 * Created by darrankelinske on 2/1/18.
 */

interface SchedulerProvider {

    fun mainThread(): Scheduler

    fun computation(): Scheduler

    fun io(): Scheduler
}